File: BACKEND/E-POS/src/main/java/com/zam/dev/pos/backend/services/TransactionService.java
package com.zam.dev.pos.backend.services;

import com.zam.dev.pos.backend.dto.requests.TransactionRequest;
import com.zam.dev.pos.backend.dto.responses.TransactionResponse;
import com.zam.dev.pos.backend.entities.Tenant;
import com.zam.dev.pos.backend.entities.User;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;

import java.time.LocalDate;

public interface TransactionService {

    TransactionResponse createTransaction(TransactionRequest request, User user);


    TransactionResponse getById(Long id);

    TransactionResponse getByUuid(String uuid);

    Page<TransactionResponse> getAll(Pageable pageable,String search , User user);

    Page<TransactionResponse> getByUserId(Long userId, Pageable pageable);

    Page<TransactionResponse> getByDateRange(
            LocalDate startDate,
            LocalDate endDate,
            User user,
            Pageable pageable
    );


    long getTotalRevenue(
            Tenant tenant
    );

    long getTotalOrders(
            Tenant tenant
    );

    void deleteByUuid(String uuid, User user);

}

File: FE/e-pos-fe/src/services/TransactionService.ts
import api from "@/lib/api";
import type { TransactionRequest, TransactionResponse } from "@/types/typeApi";

export class TransactionService {
    static async delete(uuid: string) {
        const response = await api.delete(`/transactions/${uuid}`)
        return response.data
    }


    static async getByUuid(uuid: string) {
        const response = await api.get(`/transactions/${uuid}`)
        return response.data
    }

    static async create(request: TransactionRequest): Promise<TransactionResponse> {
        const response = await api.post<TransactionResponse>('/transactions/', request);
        return response.data;
    }

    static async getAll(
        page: number,
        size: number,
        search?: string,
        sort?: string,
        startDate?: string,
        endDate?: string
    ) {
        const response = await api.get("/transactions", {
            params: {
                page,
                size,
                search: search || undefined,
                sort: sort || "createdAt,desc",
                startDate: startDate,
                endDate: endDate
            },
        });

        return response.data;
    }

}